### ISSUE TITLE
[dropdown] support vertical description field for remote data

### ISSUE THREAD

**Issue by jamessampford** [fomantic/Fomantic-UI#1845]
~~Suggest adding `unfiltered` to the remote data fields (similar to disabled)
[https://fomantic-ui.com/modules/dropdown.html#unfilterable-items-in-search-selection]~~ (pointless, since could always return the item regardless)

Also suggest to add the vertical description field (horizontal may also be useful) as well so this could be populated via the remote data as well 
[https://fomantic-ui.com/modules/dropdown.html#description]

Wonder if possible to add some kind of template field as well, so one could always set a specific layout as required to append as the item - for complete customisation, eg.

```
'template': function(result){
    return '<div class="item" data-value="' + result.id + '">' + result.text + '</div>';
}
```

This way, developers could have complete control over what could show in the dropdown - perhaps would want to a different format depending on a property of the result

It may also be desired to have a way to preserve an actionable item (#1847) so that there is a way to keep certain items in the list at all times - possibly by providing an array of classes to remove/preserve - or alternatively, and possibly better, have some kind of target path for the results to go in so headers/fixed items above/below can remain ... for example target `.scrolling.menu` as in https://fomantic-ui.com/modules/dropdown.html#search-in-menu as opposed to just the first instance of `.menu`

**Comment by lubber-de** [comment]
Beside your PR #1852, you can already modify the whole menu rendering as it itself is already a template setting.
So you can modify it according to your needs globally by

```javascript
$.fn.dropdown.settings.template.menu = menu: function(response, fields, preserveHTML, className) {
// render the menu according to your needs and return the html as string
};
```

Providing a separate templates for the single item might gets complicated as in 2.8.8 submenus are support by calling the menu template recursively.

**Comment by jamessampford** [comment]
Ah, yes, I see what you mean - perhaps rather than having it as an parameter in the list of values, we could have an additional setting that can be modified from initialising the dropdown so it could potentially be customised per dropdown to override the original template for a menu item, as opposed to setting a global customisation (if that makes sense), something like the below:

```
$('.ui.dropdown').dropdown({
  menuItemTemplate: function(item) {
    return '<div class="item" data-value="' + item.value + '">' + item.text + '</div>';
  }
});
```

Where `item` is already escaped and provides all possible fields completed with whatever data was provided and defaults where not provided - though I imagine this is the easy bit, the difficult bit could be the submenu aspect, but it could well be that the submenu wrapper is still handled by dropdown.js

Possibly should change this to its own issue

**Comment by lubber-de** [comment]
Yes, let's create a separate feature request issue for that. Once your PR #1852 is approved/merged we have a clean(er) base for a possible code separation of item rendering into a dedicated template function.
